C++ compilation warning for Z_TIMEOUT_TICKS

**Describe the bug**
Compilation of C++ code with Zephyr includes throws following compilation warning:
```
[20/137] Building CXX object CMakeFiles/app.dir/src/main.cpp.obj
In file included from /home/hid/zetst/zephyr/include/kernel_includes.h:41,
                 from /home/hid/zetst/zephyr/include/kernel.h:17,
                 from /home/hid/zetst/zephyr/include/zephyr.h:18,
                 from /home/hid/zetst/zephyr/samples/cpp_synchronization/src/main.cpp:22:
/home/hid/zetst/zephyr/include/kernel.h: In function 'int32_t k_msleep(int32_t)':
/home/hid/zetst/zephyr/include/sys_clock.h:89:62: warning: narrowing conversion of 'k_ms_to_ticks_ceil64(((uint64_t)((ms > 0) ? ms : 0)))' from 'uint64_t' {aka 'long long unsigned int'} to 'k_ticks_t' {aka 'long long int'} [-Wnarrowing]
   89 | # define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS(k_ms_to_ticks_ceil64(MAX(t, 0)))
      |                                                              ^
/home/hid/zetst/zephyr/include/sys_clock.h:85:55: note: in definition of macro 'Z_TIMEOUT_TICKS'
   85 | #define Z_TIMEOUT_TICKS(t) ((k_timeout_t) { .ticks = (t) })
      |                                                       ^
/home/hid/zetst/zephyr/include/kernel.h:940:17: note: in expansion of macro 'Z_TIMEOUT_MS'
  940 |  return k_sleep(Z_TIMEOUT_MS(ms));
      |                 ^~~~~~~~~~~~
```
The exact part which gives warning is: `.ticks = (t)`
Adding an explicit cast eliminates above warning:
`#define Z_TIMEOUT_TICKS(t) ((k_timeout_t) { .ticks = (k_ticks_t)(t) })`

**To Reproduce**
Compile any C++ code for current Zephyr codebase i.e.
west build -p auto -b nucleo_wb55rg samples/cpp_synchronization

**Expected behavior**
No compilation warning.

**Impact**
Annoyance.

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK 0.11.3
 - Commit: ba0a791635c9fa70f0dc5ffb7a98a03f7e28f000

--- a/include/sys_clock.h
+++ b/include/sys_clock.h
@@ -86,15 +86,15 @@ typedef struct {
 #define Z_FOREVER Z_TIMEOUT_TICKS(K_TICKS_FOREVER)
 
 #ifdef CONFIG_TIMEOUT_64BIT
-# define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS(k_ms_to_ticks_ceil64(MAX(t, 0)))
-# define Z_TIMEOUT_US(t) Z_TIMEOUT_TICKS(k_us_to_ticks_ceil64(MAX(t, 0)))
-# define Z_TIMEOUT_NS(t) Z_TIMEOUT_TICKS(k_ns_to_ticks_ceil64(MAX(t, 0)))
-# define Z_TIMEOUT_CYC(t) Z_TIMEOUT_TICKS(k_cyc_to_ticks_ceil64(MAX(t, 0)))
+# define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS((k_ticks_t)k_ms_to_ticks_ceil64(MAX(t, 0)))
+# define Z_TIMEOUT_US(t) Z_TIMEOUT_TICKS((k_ticks_t)k_us_to_ticks_ceil64(MAX(t, 0)))
+# define Z_TIMEOUT_NS(t) Z_TIMEOUT_TICKS((k_ticks_t)k_ns_to_ticks_ceil64(MAX(t, 0)))
+# define Z_TIMEOUT_CYC(t) Z_TIMEOUT_TICKS((k_ticks_t)k_cyc_to_ticks_ceil64(MAX(t, 0)))
 #else
-# define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS(k_ms_to_ticks_ceil32(MAX(t, 0)))
-# define Z_TIMEOUT_US(t) Z_TIMEOUT_TICKS(k_us_to_ticks_ceil32(MAX(t, 0)))
-# define Z_TIMEOUT_NS(t) Z_TIMEOUT_TICKS(k_ns_to_ticks_ceil32(MAX(t, 0)))
-# define Z_TIMEOUT_CYC(t) Z_TIMEOUT_TICKS(k_cyc_to_ticks_ceil32(MAX(t, 0)))
+# define Z_TIMEOUT_MS(t) Z_TIMEOUT_TICKS((k_ticks_t)k_ms_to_ticks_ceil32(MAX(t, 0)))
+# define Z_TIMEOUT_US(t) Z_TIMEOUT_TICKS((k_ticks_t)k_us_to_ticks_ceil32(MAX(t, 0)))
+# define Z_TIMEOUT_NS(t) Z_TIMEOUT_TICKS((k_ticks_t)k_ns_to_ticks_ceil32(MAX(t, 0)))
+# define Z_TIMEOUT_CYC(t) Z_TIMEOUT_TICKS((k_ticks_t)k_cyc_to_ticks_ceil32(MAX(t, 0)))
 #endif
 
 /* Converts between absolute timeout expiration values (packed into

